channel-interceptor/src/main/scala/fr/acinq/eclair/plugins/channelinterceptor/OpenChannelInterceptorPlugin.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2023 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.plugins.channelinterceptor
+
+import akka.actor.ActorSystem
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.scaladsl.adapter.{ClassicActorRefOps, ClassicActorSystemOps}
+import akka.actor.typed.{ActorRef, SupervisorStrategy}
+import com.typesafe.config.{Config, ConfigFactory}
+import fr.acinq.bitcoin.scalacompat.Satoshi
+import fr.acinq.eclair.{InterceptOpenChannelCommand, InterceptOpenChannelPlugin, Kit, NodeParams, Plugin, PluginParams, Setup}
+import grizzled.slf4j.Logging
+
+import java.io.File
+
+
+/**
+ * Intercept OpenChannel messages received by the node and respond by continuing the process
+ * of accepting the request, potentially with different local parameters, or failing the request.
+ */
+class OpenChannelInterceptorPlugin extends Plugin with Logging {
+  var pluginKit: OpenChannelInterceptorKit = _
+  var config: Config = _
+
+  override def params: PluginParams = new InterceptOpenChannelPlugin {
+    // @formatter:off
+    override def name: String = "OpenChannelInterceptorPlugin"
+    // @formatter:on
+    override def openChannelInterceptor: ActorRef[InterceptOpenChannelCommand] = pluginKit.openChannelInterceptor
+  }
+
+  override def onSetup(setup: Setup): Unit = {
+    // load or generate seed
+    config = loadConfiguration(setup.datadir)
```

Comment:
This is necessary because of the two-step initialization of plugins. We can only access `setup.datadir` in `OnSetup` which we need to load `config`. We can either have a `var config` or load the parameters from `config` in `OnSetup` and create a `var` to hold them until `OnKit` is called.

Reply:
Fixed the nit in [2ac483e](https://github.com/ACINQ/eclair-plugins/pull/4/commits/2ac483eb4edfcb57b81e05d6faeebf4aef6b471e)